src/algo.rs

```diff
@@ -349,4 +368,20 @@ mod tests {
         let profiles = compute(&t, &[2, 3]);
         assert_eq!(23, profiles[0][0].arr_time);
     }
+
+    #[test]
+    fn build_route() {
+        let mut b = Timetable::builder();
+        b.trip()
+            .s("a", "0:10")
+            .s("b", "0:20")
+            .trip()
+            .s("b", "0:30")
+            .s("c", "0:40");
+        let t = b.build();
+        let profiles = compute(&t, &[2]);
+        assert_eq!(1, profiles[0].len());
+        let route = profiles[0][0].route(&profiles, &t);
+        assert_eq!(2, route.len());
```

Comment:
you don't check much there :stuck_out_tongue_winking_eye:

Reply:
It was hard enough to get that to work! But I added a test to be sure the connections are in the right order